components/catalog.js

```diff
@@ -0,0 +1,79 @@
+import { DatasetCard } from '@/components/dataset-card'
+import { SearchBox } from '@/components/search-box'
+import { SampleDatasets } from '@/data/sample-datasets'
+import { Column, Row } from '@carbonplan/components'
+import { useEffect, useState } from 'react'
+import { Box, Flex, Text } from 'theme-ui'
+
+// Custom hook to get the current window width
+function useWindowWidth() {
+  const [windowWidth, setWindowWidth] = useState(
+    typeof window !== 'undefined' ? window.innerWidth : 1024, // Default to desktop width if window is not available
+  )
+
+  useEffect(() => {
+    if (typeof window === 'undefined') return // Skip if window is not available
+
+    const handleResize = () => setWindowWidth(window.innerWidth)
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
+
+  return windowWidth
+}
+
+export const Catalog = ({}) => {
+  const datasets = SampleDatasets
+  const windowWidth = useWindowWidth()
+
+  // Helper function to calculate the start and width based on the screen size
+  const getStartAndWidth = (index) => {
+    if (windowWidth >= 1024) {
+      // Desktop
+      return { start: (index % 3) * 4 + 1, width: 4 }
+    } else if (windowWidth >= 768) {
+      // Tablet
+      return { start: (index % 2) * 4 + 1, width: 3 }
+    } else {
+      // Mobile
+      return { start: 1, width: 6 }
+    }
+  }
```

Comment:
@katamartin,  🤔 am i making this too complicated?  let me know if there is a simpler way to accomplish this

Reply:
I left the exact suggestion inline, but you should be able to use breakpoint-specific array styles to avoid any bespoke "responsive" logic (i.e. manual checks against window width). As a rule, you should almost always be able to define your responsive grid styles inline! This should be true of other non-grid-related responsive styles as well.

Hopefully this feels much more manageable.